projectile.el

```diff
@@ -721,6 +721,22 @@ The cache is created both in memory and on the hard drive."
                                           projectile-cache-file))
     (projectile-invalidate-cache nil)))
 
+;;;###autoload
+(defun projectile-discover-projects-in-directory (directory)
+  "Discover any projects in DIRECTORY and add them to the projectile cache.
+This function is not recursive and only adds projects with roots
+at the top level of DIRECTORY."
+  (interactive
+   (list (read-directory-name "Starting directory: ")))
+  (let ((subdirs (directory-files directory t)))
+    (mapcar
+     (lambda (dir)
+       (when (and (file-directory-p dir)
+                  (not (member (file-name-nondirectory dir) '(".." "."))))
+         (let ((default-directory dir))
+           (projectile-cache-projects-find-file-hook))))
```

Comment:
This is not correct. You should simply check whether you're in a project dir and if so add this dir to the list of known projects.